src/economy/economy.cpp

```diff
@@ -4299,35 +6650,60 @@ void daily_update(sys::state& state, bool initiate_buildings) {
 							if(e.second > best_p.second)
 								best_p = e;
 
-						auto new_rr = fatten(state.world, state.world.force_create_province_building_construction(best_p.first, n));
+						auto new_rr = fatten(
+							state.world,
+							state.world.force_create_province_building_construction(best_p.first, n)
+						);
 						new_rr.set_is_pop_project(true);
 						new_rr.set_type(uint8_t(province_building_type::railroad));
 						//found_investment = true;
 					}
 				}
 			}
-			n.set_private_investment(0.0f);
+			//n.set_private_investment(0.0f);
```

Comment:
Do you clear the private investment amount in some other place? Because without that I believe that investment money will just increase constantly day over day.

Reply:
```cpp
			float pi_total = full_private_investment_cost(state, n);
			float perceived_spending = pi_total * 10.f;
			float pi_budget = state.world.nation_get_private_investment(n);
			auto pi_scale = perceived_spending <= pi_budget ? 1.0f : pi_budget / perceived_spending;
			//cut away low values:
			pi_scale = std::max(0.f, pi_scale - 0.1f);
			state.world.nation_set_private_investment_effective_fraction(n, pi_scale);
			state.world.nation_set_private_investment(n, std::max(0.0f, pi_budget - pi_total * pi_scale));
```

Reply:
Instead of clearing it, i reduce it by with actual spending.

Reply:
The reason we don't do that previously is because it can result in undesirable boom and bust behavior. If there is ever a situation where the investors don't have anything to build (say, because of laws) the investment pool will build up and up. Then, when they can invest, the huge pool will start many, many projects at once. Which they won't be able to actually afford day after day, and so most will sit idle clogging up spaces.

Reply:
Main reason i removed it is because i do not want investors to spend all their money on the projects as it leads to them being ready to pay any price for construction goods. If they are always eager to spend entire budget, it encourages prices to grow  up to the limit. To avoid it, I cut away lower values of private budget scale, so if price is too high and they can afford only 0.1 of goods, they spend nothing. 

Then there are two cases: this price is caused by economic environment and they save up until they can afford it. And if this high price is caused by them, they could save up until they are able to pay for 0.1 of goods and push the price further but the whole process is more smooth and allows producers and trade routes to adapt to the rising prices.

And being able to save up money is crucial for this, otherwise initially poor investors will not be able to construct anything if they decide that the price is too high for them, decide to not invest and then the game clears up their savings.

Reply:
I understand the reasoning, but then something needs to be added to prevent them from saving up simply because there is nothing to invest in. You can probably just add a boolean somewhere that detects in the possible investment section whether a no investment result is because of prices being too high, since you are testing for that, and then clear the pool if that flag is not set, meaning that excess money is accumulating for other reasons.

Reply:
I can add a process which siphons part of investment pool back to pops (not only investors, but a wide spread of pops, could represent payment for some abstract services or gifts/bribery/charity). This way it will not grow above a certain amount and will add additional money flows which the game currently lacks.

Reply:
Sounds fine to me. Just as long as we don't end up with a country full of stuck rail roads being built